psi4/src/psi4/lib3index/df_helper.cc

```diff
@@ -2186,51 +2872,58 @@ void DF_Helper::compute_J(std::vector<SharedMatrix> D, std::vector<SharedMatrix>
             rank = omp_get_thread_num();
             #endif
 
-            for (size_t m = 0, sp_count = -1; m < nao_; m++) {
+            for (size_t m = 0, sp_count = -1; m < nao; m++) {
                 if (schwarz_fun_mask_[k * nao + m]) {
                     sp_count++;
-                    C_DCOPY(1, &Dp[nao*k+m], 1, &D_buffers[rank][sp_count], 1);
+                    D_buffers[rank][sp_count] = Dp[nao*k+m];
                 }
             }
             // (Qm)(m) -> (Q)
-            C_DGEMV('N', block_size, sp_size, 1.0, &Mp[jump], sp_size, &D_buffers[rank][0], 1, 1.0, &T1p[rank*naux], 1);
+            C_DGEMV('N', block_size, sp_size, 1.0, &Mp[jump], sp_size, &D_buffers[rank][0], 1, 1.0, &T1p[rank * align_size], 1);
+        }
+        timer_off("DF_Helper~JK        - Jfirst");
+        timer_on("DF_Helper~JK       - Jreduce");
+        // reduce
+        for(size_t k=1; k<nthreads_; k++){
+            for(size_t l=0; l < naux; l++)
+                T1p[l] += T1p[k * align_size + l];
+        }
+        timer_off("DF_Helper~JK       - Jreduce");
+        timer_on("DF_Helper~JK       - Jcomple");
```

Comment:
General point, but after you rebase you will have a new timer printer. Might be worth running to see how the new scheme looks with your labeling. You may want to rejig your names slightly.